gotpm/cmd/pubkey.go

```diff
@@ -82,27 +82,13 @@ func getKey(rw io.ReadWriter, hierarchy tpmutil.Handle, algo tpm2.Algorithm) (*t
 		return tpm2tools.KeyFromNvIndex(rw, hierarchy, nvIndex)
 	}
 
-	switch algo {
-	case tpm2.AlgRSA:
-		switch hierarchy {
-		case tpm2.HandleEndorsement:
-			return tpm2tools.EndorsementKeyRSA(rw)
-		case tpm2.HandleOwner:
-			return tpm2tools.StorageRootKeyRSA(rw)
-		default:
-			return nil, fmt.Errorf("There is no default RSA key for this hierarchy")
-		}
-	case tpm2.AlgECC:
-		switch hierarchy {
-		case tpm2.HandleEndorsement:
-			return tpm2tools.EndorsementKeyECC(rw)
-		case tpm2.HandleOwner:
-			return tpm2tools.StorageRootKeyECC(rw)
-		default:
-			return nil, fmt.Errorf("There is no default ECC key for this hierarchy")
-		}
+	switch hierarchy {
+	case tpm2.HandleEndorsement:
+		return getEK(rw)
+	case tpm2.HandleOwner:
+		return getSRK(rw)
 	default:
-		panic("Unreachable")
+		return nil, fmt.Errorf("There is no default key for this hierarchy")
```

Comment:
Put the `hierarchy` in this error message.

Reply:
done